rust/ql/src/queries/unusedentities/UnusedValue.ql

```diff
@@ -20,5 +20,6 @@ where
   not write = any(Ssa::WriteDefinition def).getWriteAccess().getAstNode() and
   // avoid overlap with the unused variable query
   not isUnused(v) and
-  not v instanceof DiscardVariable
+  not v instanceof DiscardVariable and
+  not write.isInMacroExpansion()
```

Comment:
I think this restriction is fine, but it also excludes results that are in the "token tree' (arguments) of a macro call. Those tokens are assembled into AST nodes when the macro is expanded. A bit of an artificial example:
```rust
        let x;
        print!(
            "The value of x is {}",
            ({
                x = 10;
                x
            })
        )
```
With the restriction we no longer consider `10` to be an unused value.

Reply:
For `10` to be unused I think we need a bit more, perhaps:
```
x = 10;
x = 20;
x
```

It would be great to add something like this as a test case (whether or not we fix it at this stage)!

Reply:
Ah yes indeed I meant , `{ x=10; 1}`

Reply:
I have added this test (and made the test suite runnable).